### PR TITLE
77 - 404 not found page

### LINKED ISSUE #77: Create "404 not found" page

### PR DESCRIPTION
**Pull request by jKm00**
Closes #77

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -44,23 +50,32 @@ function App() {
...
-            <Nav onLogOut={handleLogOut}/>
-            <Routes>
-              <Route path="/dashboard/" element={<Dashboard/>}/>
-              <Route path="/markets" element={<Market/>}/>
-              <Route path="*" element={<Navigate to="/dashboard"/>}/>
-            </Routes>
-          </>
-          :
+      {user ? (
+        <>
```
Comment (Pipe-Runner) [inline frontend/src/App.js:54]:
```
{(
    <>
      <Nav onLogOut={handleLogOut} />
      <Routes>
       {user ? 
        (
        <>
          <Route path="/dashboard/" element={<Dashboard />} />
          <Route path="/markets" element={<Market />} />
          <Route path="*" element={<NotFound />} />
        </>
        ) : 
        (
        <>
         <Route
            path="/signin"
            element={
              <SignInUpForm isSignIn={true} onSuccess={onSignInSuccess} />
            }
          />
          <Route
            path="/signup"
            element={
              <SignInUpForm isSignIn={false} onSuccess={onSignUpSuccess} />
            }
          />
        </>
        )
      }
      </Routes>
  }
    </>
  )
}
```

This avoids repeating common routes for both branches, in this case, 404. Haven't used react-router for a while but I think this should work.

Comment (Pipe-Runner) [reply frontend/src/App.js]: Yes, that's the idea. Do verify it before merging, though. There might be caveats.

**Comment by strazdinsg** [comment]
Is it not better to redirect / to /signin if the user has not signed in already?
(instead of showing the SignIn form both on /signin and on /)

**Comment by jKm00** [comment]
> Is it not better to redirect / to /signin if the user has not signed in already? (instead of showing the SignIn form both on /signin and on /)

Yeah that's true. Will try and fix that. Was also thinking about making a welcome/landing page, but can just redirect for now

**Pull request merged**